src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -284,6 +281,11 @@ private static bool TryGetMinimalCSharpRange(RazorCodeDocument codeDocument, Ran
                 var endRange = maxGeneratedSpan.Value.AsTextSpan().AsRange(csharpSourceText);
 
                 csharpRange = new Range { Start = startRange.Start, End = endRange.End };
+                if (csharpRange.Start > csharpRange.End)
```

Comment:
Does this only happen when we can't find an end or there's no mappings?

Reply:
AFAIK this should literally never be the case (I can change it to a `Debug.Assert` to reflect that). If we "can't find an end" or there are no mappings minGeneratedSpand and/or maxGeneratedSpan will be null and we'll return no range.

Reply:
Ah, any hypothesis on how the VSTS issue was triggered?